site/app/models/gradeable/LateDayInfo.php

```diff
@@ -161,43 +236,64 @@ public function getStatusMessage() {
         }
     }
 
+    /**
+     * Gets the status messages for this gradeable
+     * @return array
+     */
+    public static function getSimpleMessageFromSatus() {
+        return [
+            self::STATUS_NO_ACTIVE_VERSION => 'No Submission',
+            self::STATUS_GOOD => 'Good',
+            self::STATUS_LATE => 'Late',
+            self::STATUS_BAD => 'Bad'
+        ];
+    }
+
     /**
      * Gets if this user has late days info available (if they have an active version)
      * @return bool
      */
     public function hasLateDaysInfo() {
-        return $this->graded_gradeable->getAutoGradedGradeable()->hasActiveVersion();
+        return $this->has_late_day_info;
     }
 
     /**
-     * Gets the number of late days charged for this assignment
+     * Gets the number of late days charged from this event
      * @return int
      */
     public function getLateDaysCharged() {
-        if ($this->getStatus() === self::STATUS_BAD) {
-            // Don't charge late days for BAD status
-            return 0;
-        }
-        // clamp the days charged to be the days late minus exceptions above zero.
-        return max(0, min($this->getDaysLate(), $this->getLateDaysAllowed()) - $this->getLateDayException());
+        return -$this->late_days_change;
+    }
+
+    /**
+     * Gets the number of late days increased from this event
+     * @return int
+     */
+    public function getLateDaysChange() {
+        return $this->late_days_change;
     }
 
     /**
      * Gets the number of days late for the active version
      * @return int
      */
     public function getDaysLate() {
-        if (!$this->hasLateDaysInfo()) {
-            return 0;
-        }
-        return $this->graded_gradeable->getAutoGradedGradeable()->getActiveVersionInstance()->getDaysLate();
+        return $this->submission_days_late;
+    }
+
+    /**
+     * Returns true if this event is a graded gradeable charge with regrades allowed
+     * @return bool
+     */
+    public function isRegradeAllowed() {
+        return $this->graded_gradeable !== null && $this->graded_gradeable->getGradeable()->isRegradeAllowed();
     }
 
     /**
      * Get number of grade inquiries pending and resolved for this gradeable
      * @return int
      */
     public function getGradeInquiryCount() {
-        return $this->graded_gradeable->getGradeInquiryCount();
+        return $this->graded_gradeable !== null ? $this->graded_gradeable->getGradeInquiryCount() : null;
```

Comment:
It would be better to use the the null coalescing (`??`) operator for this I believe.

Reply:
Since I'm checking $graded_gradeable I cant use the coalescing operator. It might be null, I wont be sure if I can use `getGradeInquiryCount()`. I can potentially shorten this with `$this->graded_gradeable?->getGradeInquiryCount() ?? null;` but we would need PHP v8 and up tp use the `?->` operator.